Cogs/AdminCommands.py

```diff
@@ -257,3 +259,51 @@ async def announce(self, ctx, *, message=''):
         await log(self.bot, f"{ctx.author} has executed the announcement command in the {ctx.channel}")
         for channel_name in msg.channel_mentions:
             await channel_name.send(message)
+
+    @commands.command()
+    @commands.has_permissions(administrator=True)
+    async def message_history(self, ctx):
+        """Outputs all messages from a specified user after a specified date and various pertinent data to a file
+        Takes in user input for the name of a user who's message history is desired and a date after which to search messages.
+        Outputs all messages from the user after the date, whether those messages are a reply, a link to those messages,
+        and all reactions to those messages to a file, which is then output to the server.
+
+        Outputs:
+            A file to chat including all messages from a user after a date, whether those messages are a reply,
+            a link to those messages, and all reactions to those messages.
+        """
+
+        guild = ctx.guild
+
+        await ctx.send(f"Please enter a user's discord username.")
+        username_message = await self.bot.wait_for("message", check=lambda message: message.author == ctx.author)
+
+        await ctx.send(f"Please enter a date in the following format: 'MM/DD/YYYY'.")
+        date_message = await self.bot.wait_for("message", check=lambda message: message.author == ctx.author)
+        date_object = datetime.strptime(date_message.content, "%m/%d/%Y")
+        
+        message_history_file = open("/tmp/message_history.txt", "w")
+        channels = guild.text_channels
+
+        for channel in channels:
+            async for message in channel.history(limit=None, after=date_object):
+                if message.author.name == username_message.content and message.type is MessageType.default:
```

Comment:
In the future, we may want to do a check against the unique identifier.

People can have the same username, but they'll always have a different `#0123`